test/Tests/src/Network/Http_Spec.enso

```diff
@@ -50,156 +51,192 @@ spec =
         base_url_with_slash = if base_url.ends_with "/" then base_url else base_url + "/"
         url_get = base_url_with_slash + "get"
 
-        Test.specify "Can perform a Get with a JSON response" <|
-            response = HTTP.fetch url_get
+        Test.specify "Can perform a GET with a JSON response" <|
+            response = Data.fetch url_get
             response.at "headers" . at "Content-Length" . should_equal "0"
 
-        Test.specify "Can perform a Get with a file response" <|
+        Test.specify "Can perform a GET with a file response" <|
             f = enso_project.data / "spreadsheet.xls"
             f.delete_if_exists
             url = "https://enso-data-samples.s3.us-west-1.amazonaws.com/spreadsheet.xls"
-            HTTP.fetch url . to_file f
+            Data.fetch url . body . to_file f
             f.size . should_equal 7168
             f.delete
             f.exists.should_be_false
 
         Test.specify "Can skip auto-parse" <|
-            response = HTTP.fetch url_get try_auto_parse_response=False
-            response.bytes.length . should_equal 272
+            response = Data.fetch url_get try_auto_parse_response=False
+            response.code.code . should_equal 200
+            response.body.bytes.length . should_equal 272
+
+        Test.specify "Can perform a HEAD" <|
+            # httpbin.org does not provide a HEAD endpoint, so we use GET
+            response = Data.fetch url_get method=HTTP_Method.Head #try_auto_parse_response=False
+            response.code.code . should_equal 200
+            response.body.bytes.length . should_equal 0
+
+        Test.specify "Can perform an OPTIONS" <|
+            # httpbin.org does not provide an OPTIONS endpoint, so we use GET
+            response = Data.fetch url_get method=HTTP_Method.Options
+            response.code.code . should_equal 200
+            response.body.bytes.length . should_equal 0
+
+        Test.specify "Can still perform request when output context is disabled" <|
+            run_with_and_without_output <|
+                Data.fetch url_get try_auto_parse_response=False . code . code . should_equal 200
+                Data.fetch url_get method=HTTP_Method.Head try_auto_parse_response=False . code . code . should_equal 200
+                Data.fetch url_get method=HTTP_Method.Options try_auto_parse_response=False . code . code . should_equal 200
```

Comment:
What if I do `Data.fetch method=HTTP_Method.Delete`? Will it work or not?

Ofc it should throw `Forbidden_Operation` in disabled context, but I think only `Data.post` is tested this way. Can we add a test case for this as well?

Reply:
There is a test for this called "Cannot DELETE through fetch".